src/libguac/protocol.c

```diff
@@ -1241,3 +1304,34 @@ int guac_protocol_decode_base64(char* base64) {
 
 }
 
+guac_protocol_version guac_protocol_string_to_version(char* version_string) {
+    
+    guac_protocol_version_mapping* current = guac_protocol_version_table;
+    while (current->version != GUAC_PROTOCOL_VERSION_UNKNOWN) {
+        
+        if (strcmp(current->version_string, version_string) == 0)
+            return current->version;
+        
+        current++;
+        
+    }
+    
+    return GUAC_PROTOCOL_VERSION_UNKNOWN;
+    
+}
+
+const char* guac_protocol_version_to_string(guac_protocol_version version) {
+    
+    guac_protocol_version_mapping* current = guac_protocol_version_table;
+    while (current->version) {
+        
+        if (current->version == version) {
+            return (const char*) current->version_string;
+        }
+        
+    }
+    
+    return NULL;
+    
+}
```

Comment:
Looks good, but I would suggest adding unit tests for these, now that our unit testing tooling is more reasonable.

https://github.com/apache/guacamole-server/blob/c10ceab7e80f9d6eae9ad98254bf97f41a0de112/README-unit-testing.md

Reply:
Okay, I *think* I've got these implemented correctly with the latest commit...